gui/default/syncthing/core/syncthingController.js

```diff
@@ -521,6 +526,19 @@ angular.module('syncthing.core')
             console.log("recalcCompletion", device, $scope.completion[device]);
         }
 
+        function replaceAddressPort(address, newPort) {
+            var lastColonIndex = 0
+            for (var index = 0; index < address.length; index++) {
+                if (address[index] === ":") {
+                    var lastColonIndex = index
+                }
+            }
+            if (newPort === "") {
+                return address.substr(0, lastColonIndex)
+            }
+            return address.substr(0, lastColonIndex + 1) + newPort
```

Comment:
What if there is no colon, this will end up substr(0, 0+1), returning something not very sensible?

Reply:
I thought it would always have the format of ip-address:port. if you believe it could also be situation where the port does not exist. Maybe I can just default the last colon index to the end of the string. That way if it never sees a colon, it will just return the same string. Does this solution solve your question because I am not sure if are talking about?